cmake/modules/root.cmake

```diff
@@ -26,6 +26,7 @@ zephyr_file(APPLICATION_ROOT MODULE_EXT_ROOT)
 
 # Convert paths to absolute, relative from APPLICATION_SOURCE_DIR
 zephyr_file(APPLICATION_ROOT BOARD_ROOT)
+set(BOARD_ROOT ${BOARD_ROOT} CACHE PATH "Board root directories")
```

Comment:
actually this doesn't address the concern I had when writing this comment:
https://github.com/zephyrproject-rtos/zephyr/pull/53816#discussion_r1071250925

because as I also write, only paths given with `-D`, meaning they are in the CMake cache should be converted to absolute. This is what happens here: https://github.com/zephyrproject-rtos/zephyr/blob/6dc20d5ced9e88708cee8d7266b228b8e90968e2/cmake/modules/extensions.cmake#L2185-L2191

so setting `BOARD_ROOT` in cache if it already in the CMake cache is not changing anything.

But how does a sysbuild invoked build with a `-DBOARD_ROOT=<relative_path>` behave when being passed from sysbuild to the image specific build ?
Especially if multiple images are included in a build, like `hello_world` and `mcuboot`.
Will they both calculate the same absolute path based on above relative setting ?
Or will they end-up calculating the path based on their own source location, i.e. `hello_world` adds one `BOARD_ROOT` to list, but `mcuboot` adds another.

Reply:
I tested this with hello_world with sysbuild whereby hello_world had a sysbuild.cmake to include mcuboot on another board, with 2 custom boards in different paths, both specified with relative paths from the build directory. Without the changes, it fails because the path is passed to hello_world unchanged, then it fails because from the hello_world path, it is not valid, with these changes, both projects are configured correctly.

From zephyr/samples/hello_world/_CC:
```
rm -rf * && cmake -GNinja -DBOARD=custom_plank -DAPP_DIR=.. "-DBOARD_ROOT=../../../../example-application/;../../../../other" /tmp/aa/zephyr/share/sysbuild/

   *********************************
   * Running CMake for hello_world *
   *********************************

Loading Zephyr default modules (Zephyr repository).
-- Application: /tmp/aa/zephyr/samples/hello_world
-- CMake version: 3.25.1
-- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.9") found components: Interpreter 
-- Cache files will be written to: /home/jamie/.cache/zephyr
-- Zephyr version: 3.2.99 (/tmp/aa/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
-- Board: custom_plank
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.15.0 (/opt/zephyr-sdk-0.15.0)
-- Found toolchain: zephyr 0.15.0 (/opt/zephyr-sdk-0.15.0)
-- Found Dtc: /opt/zephyr-sdk-0.15.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /tmp/aa/example-application/boards/arm/custom_plank/custom_plank.dts

...

   ******************************
   * Running CMake for mcuboot2 *
   ******************************

Loading Zephyr default modules (Zephyr workspace).
-- Application: /tmp/aa/bootloader/mcuboot/boot/zephyr
-- CMake version: 3.25.1
-- Found Python3: /usr/bin/python3.10 (found suitable exact version "3.10.9") found components: Interpreter 
-- Cache files will be written to: /home/jamie/.cache/zephyr
-- Zephyr version: 3.2.99 (/tmp/aa/zephyr)
-- Found west (found suitable version "0.14.0", minimum required is "0.7.1")
-- Board: bob
-- ZEPHYR_TOOLCHAIN_VARIANT not set, trying to locate Zephyr SDK
-- Found host-tools: zephyr 0.15.0 (/opt/zephyr-sdk-0.15.0)
-- Found toolchain: zephyr 0.15.0 (/opt/zephyr-sdk-0.15.0)
-- Found Dtc: /opt/zephyr-sdk-0.15.0/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /tmp/aa/other/boards/arm/bob/bob.dts
```

Reply:
> This is not valid because relative paths can only be supplied on the command line using -D

exactly, so if I do
```
west build ... --sysbuild <abspath>/samples/foo -- -DSB_BOOTLOADER_MCUBOOT=y -DBOARD_ROOT=foo_boards
```
then the location `<abspath>/samples/foo/foo_boards` as `BOARD_ROOT` should be used for both the `foo` sample and `mcuboot`.
My worry was that if just the relative path is passed from sysbuild to all images, then `mcuboot` may calculate the abspath to become `<workspace>/bootloader/mcuboot/boot/zephyr/foo_boards` and not the path above.
Or that it would fail the CMake for foo sample / mcuboot, because CMake invocation cannot see that this variable was provided by the user using `-D` when comming from sysbuild.

Based on how CMake generally behaves i'm still concerned about this line of code and it's behavour.